### ISSUE TITLE
Add custom workouts

### ISSUE THREAD

**Issue by flomerz** [zoffline/zwift-offline#199]
Is it possible to add custom workouts by file?
I have only an android client running. I tried to add zwo files to the customworkouts folder and edited the workouts.files file but i am struggling with the checksum. Any idea how it is calculated?

**Comment by oldnapalm** [comment]
Just adding to customworkouts folder won't work, you also need to add to the database.

https://github.com/zoffline/zwift-offline/blob/fd6ad987659a6c9839fd28491b6306a24a0665f1/zwift_offline.py#L1322

No idea about the checksum, I thought it could be 8 bit (`sum(open(filename, 'rb').read()) % 256`) but it's not, and there's also the guid.

**Comment by flomerz** [comment]
I'm having trouble importing my workouts from intervals.icu. I've tried various methods, but so far, I haven't been successful.

**Comment by oldnapalm** [comment]
Have you tried this?

https://forum.intervals.icu/t/upload-workouts-to-dropbox-for-zwift-etc/25190

You need a Windows or Mac device though.

**Comment by flomerz** [comment]
Yes, that would work. But i normally use an android tablet and there is no
workouts folder as far as i know?

oldnapalm ***@***.***> schrieb am Fr., 27. Okt. 2023, 13:41:

**Comment by oldnapalm** [comment]
> But i normally use an android tablet and there is no workouts folder as far as i know?

No, there is not, that's why you need a Windows or Mac device. Once the workout has been uploaded to zoffline by the Windows/Mac client, it should appear on the Android device.

Quoting David from Intervals.icu

> Note that if you also use Zwift on iOS or other devices Zwift cloud sync will make the workouts show up there as well. So even if you don’t use Zwift on a Mac or Windows computer you can download and install the app just to enable Dropbox sync.

**Comment by flomerz** [comment]
Ok, thans for your reply. It's bad for me since I only use Linux.
But I understand that the direct sync (intervals.icu -> zoffline) is a lot
more complicated to implement.
The main problem is the unknown checksum calculation. Once this is figured,
the direct sync should be easy to implement and I would like contribute to
implement it.

**Comment by fatsbrown** [comment]
> Yes, that would work. But i normally use an android tablet and there is no
> workouts folder as far as i know?

I know it's been almost 3 years, but if you are still interested, give [this](https://github.com/fatsbrown/zwift-offline/tree/workouts) a try. Thanks to @sumulige who implemented both Intervals.icu today's workout sync and workouts.files management. I just changed to maintain the catalog on server side, so it should work with Android client too.
